core/lib/makeswift/client.ts

```diff
@@ -1,11 +1,34 @@
 import { Makeswift } from '@makeswift/runtime/next';
+import { getSiteVersion } from '@makeswift/runtime/next/server';
 import { strict } from 'assert';
+import { getLocale } from 'next-intl/server';
 
-import { runtime } from '~/lib/makeswift/runtime';
+import { defaultLocale } from '~/i18n/routing';
+
+import { runtime } from './runtime';
 
 strict(process.env.MAKESWIFT_SITE_API_KEY, 'MAKESWIFT_SITE_API_KEY is required');
 
 export const client = new Makeswift(process.env.MAKESWIFT_SITE_API_KEY, {
   runtime,
   apiOrigin: process.env.MAKESWIFT_API_ORIGIN,
 });
+
+export const getPageSnapshot = async ({ path, locale }: { path: string; locale: string }) =>
+  await client.getPageSnapshot(path, {
+    siteVersion: await getSiteVersion(),
+    locale: normalizeLocale(locale),
+  });
```

Comment:
This function can only be called on a React Server Component or Route Handler since it uses `getSiteVersion`. We might want to inline this in the `<Page>` RSC so users don't mistakenly try to use this elsewhere. What do you think?

Reply:
@migueloller Are you saying that they might see `getPageSnapshot` and try to call it instead `client.getPageSnapshot` in contexts outside of rendering? I can theoretically see that happening, but I don't think it's an issue worth sacrificing the function for. It mirrors `getComponentSnapshot` and I like that all of our uses of `getSiteVersion` and `normalizeLocale` are self-contained in the single file. If somewhere were to copy & paste the `lib/maskeswift` dir in a different project, potentially w/ a different router, this would be the only file they need to change.

Reply:
I think that all uses should be hidden behind a `<Page>` or `<MakeswiftComponent>` RSC anyway. There should be no need to call `getPageSnapshot` or `getComponentSnapshot` outside those two RSCs

Reply:
Also, if this were to be copy/pasted to a Pages Router app, it would be incompatible. If it's App Router, then my previous comment stands: the RSCs can be the API, hiding `getComponentSnapshot` and `getPageSnapshot` completely

Reply:
> Also, if this were to be copy/pasted to a Pages Router app, it would be incompatible

Yes, but the incompatibility/point of variability would be limited to a single file rather than spread through the dir.

> I think that all uses should be hidden behind a <Page> or <MakeswiftComponent> RSC anyway. There should be no need to call `getPageSnapshot` or `getComponentSnapshot` outside those two RSCs

That's true. We don't have a userland `<MakeswiftComponent>` right now; I'll make one/refactor the `getComponentSnapshot` usage once https://github.com/bigcommerce/catalyst/pull/1783 is merged.